### PR TITLE
multlabel prediction atmoml (uncertainity weighting)

### PR DESCRIPTION
**Pull request by VinayVL20**
Fixed typos, enhance model architecture, and improve visualization functions

- Corrected variable name from NUM_SAPTIAL_FEATURES to NUM_SPATIAL_FEATURES in constants.py.
- Adjusted LSTM dropout rates in model.py for better performance.
- Implemented uncertainty-weighted loss function in model.py to improve task performance.
- Added new metrics and visualization functions in visualizer.py for better analysis of task balance and prediction statistics.
- Enhanced plotting functions to provide clearer insights into training metrics and task performance.

### CODE REVIEW

File: usl_models/usl_models/atmo_ml/model.py
Comment (ahmustafa) [inline usl_models/usl_models/atmo_ml/model.py:1]: @VinayVL20 why changed dropout to 0.1 instead of 0.2. lowering dropout from may reduce regularization and increase overfitting risk- right?

File: usl_models/usl_models/atmo_ml/model.py [hunk 1]
```diff
@@ -183,27 +187,70 @@ def hypermodel(hp: keras_tuner.HyperParameters):
...
+            # s_i = log_vars[i] - log(prior_i)  (clipped for stability)
+            s = tf.clip_by_value(log_vars - tf.math.log(priors), -2.0, 2.0)
+
+            # ½·e^(−s)·L + ½·s
+            weighted = 0.5 * tf.exp(-s) * task_losses + 0.5 * s
+            return tf.reduce_sum(weighted)
+
+        # ----- Optimizer -----
+        optimizer = (
+            keras.optimizers.Adam(learning_rate=5e-4, clipnorm=1.0)
```
[the remark this answers is not in the document]
Comment (VinayVL20) [reply usl_models/usl_models/atmo_ml/model.py:242]: thanks for the insight ahmed, i think this approach would work much better. as you said. will be changing the optimizer part.
